File: leetcode/editor/en/[1282]Group the People Given the Group Size They Belong To.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} groupSizes
 * @return {number[][]}
 */
var groupThePeople = function (groupSizes) {
    const res = [];
    let map = new Map();
    for (let i = 0; i < groupSizes.length; i++) {
        const size = groupSizes[i];
        if (!map.has(size)) {
            map.set(size, []);
        }
        map.get(size).push(i);
        if (map.get(size).length === size) {
            res.push(map.get(size));
            map.set(size, []);
        }
    }
    return res;
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1282]Group the People Given the Group Size They Belong To.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        List<List<Integer>> res = new ArrayList<>();
        Map<Integer, List<Integer>> map = new HashMap<>();
        for (int i = 0; i < groupSizes.length; i++) {
            int size = groupSizes[i];
            if (!map.containsKey(size)) {
                map.put(size, new ArrayList<>());
            }
            map.get(size).add(i);
            if (map.get(size).size() == size) {
                res.add(map.get(size));
                map.put(size, new ArrayList<>());
            }
        }
        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
